xbout/load.py

```diff
@@ -126,6 +129,17 @@ def open_boutdataset(datapath='./BOUT.dmp.*.nc', inputfilepath=None,
     if geometry:
         if info:
             print("Applying {} geometry conventions".format(geometry))
+
+        if gridfilepath is not None:
+            print('here in load', ds.attrs)
+            ds.attrs["_grid"] = open_boutdataset(gridfilepath, chunks=chunks,
```

Comment:
If we store the grid data in the attrs like this then the dataset won't be serializable to netcdf. If the grid variables are just normal variables then it would be fine.

Reply:
The `_grid` is possibly only needed temporarily - the current use is that some variables are read from it and put into the dataset in `add_toroidal_geometry_coords()` and `add_s_alpha_geometry_coords()`. I'm inclined to leave it available once it's added in case we want to get other info out of it later, but for the way it's being used here we could happily just drop/ignore the `_grid` dataset when saving to netcdf, as any needed variables have been copied to the main dataset.

Reply:
...so the way I've implemented so far isn't a good idea, as @TomNicholas says. Suggestions for a replacement welcome!

Reply:
I've fixed this problem by making the `_grid` dataset a member of the `BoutDataset` accessor, rather than being in `attrs`, and added saving to netCDF to the tests of `'toroidal'` and `'s-alpha'` geometries.

I found a problem, which might well be an issue to address later: a variable that's added as a 'coordinate' shouldn't have metadata, or when saving to netCDF there will be an error because of not being able to save a `dict` to netCDF. Adding the variable as a coordinate is the only thing I currently need to do with the variables read from the grid file, so I've used `_open_grid` instead of `open_boutdataset` to open the grid without adding metadata. That's OK, but what if:
1. we want to use a variable from the grid file as something other than a coordinate
    - could be useful to have a `bout.addVariable` method, that adds metadata to the variable as well as adding it? 
2. we want to use a variable from, or calculated from, the dataset as a coordinates
    - do we need an `addCoordinates()` method that would strip metadata and then call `set_coords()`?